api/v1/atlasipaccesslist_types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2025 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/kube"
+)
+
+func init() {
+	SchemeBuilder.Register(&AtlasIPAccessList{}, &AtlasIPAccessListList{})
+}
+
+// AtlasIPAccessListSpec defines the desired state of AtlasIPAccessList.
+type AtlasIPAccessListSpec struct {
+	// Project is a reference to AtlasProject resource the user belongs to
+	// +kubebuilder:validation:Optional
+	Project *common.ResourceRefNamespaced `json:"projectRef,omitempty"`
+	// ExternalProject holds the Atlas project ID the user belongs to
+	// +kubebuilder:validation:Optional
+	ExternalProject *ExternalProjectReference `json:"externalProjectRef,omitempty"`
+	// Local credentials
+	api.LocalCredentialHolder `json:",inline"`
```

Comment:
Please wait for and rebase after #1971 so this can use the dual reference struct and the reconciler common code.

Reply:
Holding this PR  for this rebase